script/download-node.js

```diff
@@ -81,10 +81,10 @@ var downloadNode = function(version, done) {
   }
 
   if (fs.existsSync(filename)) {
-    getInstallNodeVersion(filename, function(error, installedVersion) {
+    getInstallNodeVersion(filename, function(error, installedVersion, installedArch) {
       if (error != null) {
         done(error);
-      } else if (installedVersion !== version) {
+      } else if (installedVersion !== version || installedArch !== process.arch) {
```

Comment:
If the bundled `node` is the wrong architecture, would that fall into the error handling above instead of reaching here?

Reply:
If the architectures are incompatible, yeah. And that'll be the same behavior as it was before this pull request, which I think is fine. But if the architectures are compatible, for example x86 on x64, then this check will be reached. Perhaps we can instead make the above error block re-download Node as well.

I'd really like to get this in to unblock atom/atom#19189, so I'm going to merge this, but let me know what your thoughts are about handling errors :).

Reply:
The alternatives I could see are:
* Download in the error handler, if it hasn't already done so
* Attempt to parse the error string to see if it's necessary
* Check file signatures to see if it needs to be replaced

I'd say the first is the only one that looks like it might be useful, but leaving it as is is probably fine!